### ISSUE TITLE
Keyboard shortcut editor documentation. 

### ISSUE THREAD

**Issue by Carreau** [jupyter/notebook#2153]
Review and document how to set shortcuts with space, comma, dash, minus, esc .... etc as they are special characters and need to be spelled out.

**Comment by gnestor** [comment]
@Carreau Would you like to tackle this before we release 5.0.0beta1 today?

**Comment by Carreau** [comment]
> @Carreau Would you like to tackle this before we release 5.0.0beta1 today?

I can try, but I have a couple things that have been requested to be done fast. 
Also, as I know how to do it, I might miss the main point on how to explain it. 
I'll try to give it a shot though.

**Comment by gnestor** [comment]
Ok, if you explain here then I can make the changes and submit a PR 👍

**Comment by Carreau** [comment]
Most of the info are already on the dialog: 

```
This dialog allows you to modify the keymap of the command mode, and persist the changes. You can define many type of shorctuts and sequences of keys.
Use dashes - to represent keys that should be pressed with modifiers, for example Shift-a, or Ctrl-;.
Separate by commas if the keys need to be pressed in sequence: h,a,l,t.
You can combine the two: Ctrl-x,Meta-c,Meta-b,u,t,t,e,r,f,l,y. Casing will have no effects: (e.g:  ; and : are the same on english keyboards). You need to explicitelty write the Shift modifier. Valid modifiers are Cmd, Ctrl, Alt ,Meta, Cmdtrl. Refer to developer docs for the corresponding keys depending on the platform.You can hover on the name/description of a command to see its exact internal name and differentiate from actions defined in various plugins. Changing the keybindings of edit mode is not yet possible.
```

things that need to be added: Modifier are (IIRC) case sensitive. Cmdtrl is Command on mac, ctrl on windows/linux. Meta is Alt/Option (IIRC). space comma dash/minus need to be explicitly spelled out.

**Comment by michaelpacer** [comment]
I can work on this today, it happened to arise in talking with @Carreau about documentation and keyboard shortcuts, so I'll be looking at this feature today anyway.

**Comment by gnestor** [comment]
Thanks @michaelpacer

**Comment by michaelpacer** [comment]
What does "close pager" mean?

Use case: wanted to map <kbd>⌃</kbd>+<kbd>[</kbd> to the equivalent of <kbd>Esc</kbd>. The only command escape is mapped onto is "close pager". Aside from being intrinsically unclear, this suggested (to me) that all the functions of escape were being mapped onto "close pager" and so adding <kbd>⌃</kbd>+<kbd>[</kbd> should have worked to allow you to (for example) exit from editing mode to command mode. It doesn't allow that.

**Comment by michaelpacer** [comment]
In the quickhelp menu, we have a explanation of the meaning of different modifier keys, do we want to add this to the keyboard shortcut editor as well? It seems like it would be worthwhile. 

Is there a reason it's `-` rather than `+` to delimit combinations?

**Comment by michaelpacer** [comment]
What does "ignore" mean?

**Comment by Carreau** [comment]
> What does "ignore" mean?

Probably ignored. Implementation detail. Is it for Shift ? It means that just pressing Shift will do nothing. 



> Is there a reason it's - rather than + to delimit combinations?

Cause that's standard (Vim, Atom, Emacs) along with no delimiter. 
The difference is tha instead of comma atom use space. There are some slight variation though. 

Esc is hardcoded (and have to be hardcoded) in many places. because it has other side effects on many browsers.

**Comment by Carreau** [comment]
binding ^[  to close pager works for me. (I have to use `Ctrl` in the definition though).

**Comment by michaelpacer** [comment]
What does "close pager" mean though?

**Comment by takluyver** [comment]
The pager is the pane at the bottom when you run e.g. 'zip?'. With that
info, I imagine you can work out what closing it is. ;-)

**Comment by takluyver** [comment]
Don't mind me, I replied before seeing it was explained elsewhere.

**Comment by takluyver** [comment]
Closing this as resolved by #2198. If anyone thinks there are more things we should improve - especially if you think there are changes we need for 5.0 - please say so.
